tests/e2e/node/unmanaged_node.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/consts"
+	"sigs.k8s.io/cloud-provider-azure/tests/e2e/utils"
+)
+
+var _ = Describe("Unmanaged nodes", Label(utils.TestSuiteUnmanagedNode), func() {
+	var (
+		cs clientset.Interface
+	)
+
+	BeforeEach(func() {
+		var err error
+		cs, err = utils.CreateKubeClientSet()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		cs = nil
+	})
+
+	It("unmanaged Nodes should not be removed", func() {
+		unmanagedLabel := map[string]string{consts.ManagedByAzureLabel: "false"}
+		nodeName := "fake-node"
+		utils.Logf("Creating a fake Node %q", nodeName)
+		node := utils.CreateNodeManifest(nodeName, unmanagedLabel)
+		_, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+		defer func() {
+			err := utils.DeleteNodes(cs, []string{nodeName})
+			Expect(err).NotTo(HaveOccurred())
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		utils.Logf("Waiting for Node %q status to change", nodeName)
+		time.Sleep(20 * time.Second)
```

Comment:
Is it possible that the node status does not change after 20s? Better if we poll to check the state.

Reply:
The expected behavior is after a while, the Node is not removed (NotReady). But I don't know how to do this better. The log is confusing and I will change it.

Reply:
Then why don't you just check the status in the following poll and remove the sleep?

Reply:
Yes, I think we can do this but I didn't do the poll logic like this before:
```
err := poll() {
  if node not exist
    then return true, nil
  return false, nil
}
if err == timeout
  then return success
if err == nil
  then return error
```

Reply:
Updated